ambry-api/src/main/java/com/github/ambry/compression/Compression.java

```diff
@@ -89,136 +78,6 @@ public interface Compression {
-   * @param sourceBufferOffset Offset in sourceBuffer to start reading.
-   * @param sourceDataSize Size of source data to read in bytes, not the size of sourceBuffer.
-   * @return Pair that contains the compressed buffer and the buffer usage sourceDataSize in bytes.
-   * @throws CompressionException Throws this exception if compression has internal failure.
-   */
-  default Pair<Integer, byte[]> compress(byte[] sourceBuffer, int sourceBufferOffset, int sourceDataSize)
-      throws CompressionException {
-    Utils.checkNotNullOrEmpty(sourceBuffer, "sourceBuffer cannot be null or empty.");
-    int compressedBufferSize = getCompressBufferSize(sourceDataSize);
-    byte[] compressedBuffer = new byte[compressedBufferSize];
-    int compressedSize = compress(sourceBuffer, sourceBufferOffset, sourceDataSize, compressedBuffer, 0, compressedBuffer.length);
-    return new Pair<>(compressedSize, compressedBuffer);
-  }
-
-  /**
-   * Decompress portion of the shared buffer in {@code compressedBuffer}.  Both the {@code compressedBuffer} and
-   * {@code sourceData} may be shared buffers and that's why both provide their offset and size as parameters.
-   * The compressed buffer is the output from compress() that contains the version, algorithm name, original data size,
-   * and compressed binary.
-   *
-   * @param compressedBuffer The buffer that contains compressed data generated by the compress() method.
-   * @param compressedBufferOffset Offset in the compressedBuffer to start reading.
-   * @param compressedDataSize Number of bytes to read in compressedBuffer.
-   * @param decompressedBuffer The buffer to hold the decompressed/original data.  This buffer may be shared.
-   * @param decompressedBufferOffset Offset in the sourceData buffer to write.
-   * @param decompressedDataSize Size of the original data in bytes, not the size of decompressedBuffer.
-   * @return The original data size which is same as number of bytes written to sourceData buffer.
-   * @throws CompressionException Throws this exception if decompression has internal failure.
-   */
-  int decompress(byte[] compressedBuffer, int compressedBufferOffset, int compressedDataSize,
-      byte[] decompressedBuffer, int decompressedBufferOffset, int decompressedDataSize) throws CompressionException;
-
-  /**
-   * Decompress the compressed buffer.  The compressedBuffer is not shared and contains compressed buffer only.
-   * This method allocates and returns the buffer to hold the decompressed data.
-   * @param compressedBuffer The compressed buffer generated in compress() method.
-   * @return The original/decompressed data.
-   * @throws CompressionException Throws this exception if decompression has internal failure.
-   */
-  default byte[] decompress(byte[] compressedBuffer) throws CompressionException {
-    Utils.checkNotNullOrEmpty (compressedBuffer, "compressedBuffer cannot be null or empty.");
-    return decompress(compressedBuffer, 0, compressedBuffer.length);
```

Comment:
This PR expanded after first review due to these deletions. Was this intended or accidental push ?